pydantic_ai_slim/pydantic_ai/agent/abstract.py

```diff
@@ -1035,6 +1070,10 @@ async def main():
             model_settings: Optional settings to use for this model's request.
             usage_limits: Optional limits on model request count or token usage.
             usage: Optional usage to start with, useful for resuming a conversation or agents used in tools.
+            metadata: Optional metadata to attach to this run. Accepts a dictionary or a callable taking
+                [`RunContext`][pydantic_ai.tools.RunContext]. The resolved dictionary is shallow merged into the
+                agent's metadata (or any [`Agent.override`][pydantic_ai.agent.Agent.override]) with run-level keys
```

Comment:
See above, this behavior is inconsistent with existing expectations, so the behavior and docstring will need updating